Managers/FrameMan.cpp

```diff
@@ -159,111 +155,123 @@ namespace RTE {
 
 /////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
-	void FrameMan::ValidateResolution(int &resX, int &resY, int &resMultiplier) {
-		if (resX * resMultiplier > m_ScreenResX || resY * resMultiplier > m_ScreenResY) {
+	void FrameMan::ValidateResolution(int &resX, int &resY, int &resMultiplier) const {
+		bool settingsNeedOverwrite = false;
+
+		if (resX * resMultiplier > m_MaxResX || resY * resMultiplier > m_MaxResY) {
+			settingsNeedOverwrite = true;
+			resX = m_MaxResX / resMultiplier;
+			resY = m_MaxResY / resMultiplier;
 			ShowMessageBox("Resolution too high to fit display, overriding to fit!");
-			resX = m_NewResX = m_ScreenResX / resMultiplier;
-			resY = m_NewResY = m_ScreenResY / resMultiplier;
 		} else if (!m_ForceDedicatedFullScreenGfxDriver && resX * resMultiplier == 1366 && resY * resMultiplier == 768) {
-			ShowMessageBox("Unfortunately, 1366x768 resolution is not supported in windowed or borderless mode. 1360x768 will be used instead!\nTo enable the use of this resolution, please force the dedicated fullscreen driver through \"Settings.ini\".");
-			resX = m_NewResX = 1360 / resMultiplier;
-			resY = m_NewResY = 768 / resMultiplier;
+			settingsNeedOverwrite = true;
+			resX = 1360 / resMultiplier;
+			resY = 768 / resMultiplier;
+			const char *invalidResolutionMessage = {
+				"Unfortunately, 1366x768 resolution is not supported in windowed or borderless mode. 1360x768 will be used instead!\n"
```

Comment:
You can't concat C strings with `+`.
When defining it like that it's concated at compile time.
Alternatively I could do
```
const char *a = " \
  peepee\n \
  poopoo   \
";
```
But that's garbage.
I remember wanting this to be a string literal but I had issues with escape sequences.

Anyway yes, this works, I wouldn't do it if it didn't. It also works without the brackets, but it's just cleaner that way.
We already had this discussion last year.

Reply:
Last year is a whole year ago so I don't remember it lol.
What was going wrong with using std::string? Would be nice to not use C strings if we can.

Reply:
Nothing really, it's just long af.
But it appears I can do this exact same thing with `std::string` as well so I guess i'll just change the type and nothing else lel.

Reply:
Okay yeah, this is java haunting me, and I think I'm starting to recall having this conversation in the past lol. In java you need to have that + to concat strings over multiple lines, in cpp you can just put em next to eachother.
I don't really like the {}s, but I also don't really like 2 strings just sitting next to eachother, so I guess do whatever.

Btw here was my test string to refresh my memory, for your appreciation:
```cpp
		std::string testString = "This string is so damn long that I don't want it on line. It just goes on and on and on and on and on and on and on and on and on.\n"
			"So I put it on two lines. Now it can fit on screen without a horizontal scrollbar that you hate so much. Something about dinosaurs I guess.";
```